internal/components/datasource/schema.go

```diff
@@ -105,7 +107,33 @@ type DataSource struct {
 	//AccessControl map[string]bool `json:"accessControl,omitempty"`
 }
 
-func (ds *DataSource) DeepCopyObject() runtime.Object {
+// TODO this should be generated by thema
+type DataSourceStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+// TODO this should be generated by thema
+type DataSource struct {
```

Comment:
...and call this one that we pass to k8s systems `DataSourceCR`.

We've a ways to go yet and i may change my take as we do and learn more, but it seems to me that this one is the more internal implementation detail, and the other is the real object that people are interested in thinking about/working with.

(strong opinion, loosely held)